scripts/SREP/infra-node-resize/script.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+# Infra-node-resize
+
+set -e
+set -o nounset
+set -o pipefail
+
+# fetch the infra machinepools
+get_machinepool() {
+    echo 'FETCHING INFRA MACHINEPOOLS...'
+    NAMESPACE=$(oc get ns | grep "$CLUSTER_ID" | awk '{print $1}')
+    oc -n "$NAMESPACE" get machinepool
+    INFRA_MACHINEPOOL=$(oc -n "$NAMESPACE"  get machinepool -o json | jq -r '.items[] | select(.spec.name=="infra") | .metadata.name')
+    oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r ".spec.platform.type"
+}
+
+# update the yaml file
+update_cluster_infra_yaml() {
+    echo 'UPDATING THE YAML FILE FOR TEMPORARY MACHINEPOOL...'
+    oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o yaml > /tmp/cluster-infra.yaml
+    yq -i 'del(.metadata.creationTimestamp) | del(.metadata.finalizer) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.metadata.uid) | del(.status)' /tmp/cluster-infra.yaml
+    cat /tmp/cluster-infra.yaml 
+    CLOUD=$(oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r '.spec.platform | keys_unsorted[0]')
+    yq -i ".metadata.name |= . + 2 | .spec.name |= . + 2 | .spec.platform.$CLOUD.type |= env(INSTANCE_SIZE)" /tmp/cluster-infra.yaml 
+}
+
+# create the temporary machinepool
+create_temp_machinepool() {
+    echo 'CREATING TEMPORARY MACHINEPOOL...'
+    oc -n "$NAMESPACE" create -f /tmp/cluster-infra.yaml
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not correctly provisioned'
+        sleep 10
+    done
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# delete the original machinepool
+delete_original_machinepool() {
+    echo 'DELETING THE ORIGINAL MACHINEPOOL...'
+    oc scale --replicas 0 machinepool "$INFRA_MACHINEPOOL" -n "$NAMESPACE"
+    oc -n "$NAMESPACE" get machinepool
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not properly deleted'
+        sleep 10
+    done
+    oc -n "$NAMESPACE" delete machinepool "$INFRA_MACHINEPOOL"
+    echo 'Original machinepools which are not resized are deleted'
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# update yaml file cluster 
+update_new_infra_yaml() {
+    echo 'UPDATING YAML FILE...'
+    yq eval '.metadata.name |= sub("2$"; "") | .spec.name |= sub("2$"; "")' -i /tmp/cluster-infra.yaml 
+    cat /tmp/cluster-infra.yaml
+}
+
+# create new machinepool
+create_new_machinepool() {
+    echo 'CREATING NEW MACHINEPOOL...'
+    oc -n "$NAMESPACE" create -f /tmp/cluster-infra.yaml
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
```

Comment:
I can see you use a `while loop with sleep` for waiting on machine provisioning and deletion. This could be a bit inefficient, since the loop keeps running as long as the condition `oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')` is true.  

In cases where machine provisioning or deletion takes longer than expected, or if there's an issue that prevents the condition from being achieved, the loop will continue running. This can lead to unexpectedly long wait times without any feedback to the user about what's happening ?  

So, may be better to introduce a timeout mechanism in **a separate function** to limit the number of checks to prevent the possible long waiting, WDYT ? cc: @bmeng

Reply:
> So, may be better to introduce a timeout mechanism in a separate function to limit the number of checks to prevent the possible long waiting, also, adding error handling with an error message when timeout WDYT ? cc: @bmeng

Thanks, Shawn for the review! I can add a check function which will catch if there's any error but usually to get into ready state I had observed that it takes 10 mins. I can set the time-limit according to that. Thanks Again :)

Reply:
`timeout` can come in handy.